purchases/src/main/kotlin/com/revenuecat/purchases/PostTransactionWithProductDetailsHelper.kt

```diff
@@ -48,10 +49,30 @@ internal class PostTransactionWithProductDetailsHelper(
                                     product.id == transaction.productIds.firstOrNull()
                                 }
                             }
+
+                        val subscriptionOptionsForGoogleProductIDs = transaction
```

Comment:
When we generate the ReceiptInfo in `postReceiptHelper.postTransactionAndConsumeIfNeeded()`, we need a way to get the subscription options for the add-on purchases so they can be included when we post the receipt to the backend. Building this `subscriptionOptionsForGoogleProductIDs` map here allows us to do that :)

Reply:
Hmm just a note that this might not work when restoring purchases... How critical is it to have this information? Just wondering if there would be issues in the case of purchasing one of these multi line subscriptions but the request failing, then us retrying to post it on next app open without this information.